lib/features/Auth/signup/presentation/pages/signup_screen.dart

```diff
@@ -24,86 +30,80 @@ class SignupScreen extends StatelessWidget {
             padding:
                 const EdgeInsets.symmetric(horizontal: 20).copyWith(top: 30),
             child: Form(
+              key: formKey,
+              // autovalidateMode: AutovalidateMode.onUserInteraction,
               child: Column(
                 crossAxisAlignment: CrossAxisAlignment.start,
                 children: [
                   CustomTextField(
-                      textEditingController: TextEditingController(),
+                      textEditingController: cubit.emailController,
                       placeholderText: "Email",
                       icon: Icons.email,
-                      isPassword: false),
+                      passwordVisibiltyIcon: false,
+                      isobscure: false,
+                      onChange: (email) => cubit.validateEmail(email)),
+
+                  BlocBuilder<SignupCubit, SignupState>(
```

Comment:
why u need BlocBuilder here ?

Reply:
for rebuild email field , but i will remove it and make BlocBuilder for entire screen